lms/extensions/feature_flags/_providers.py

```diff
@@ -26,8 +24,9 @@ def config_file_provider(request, feature_flag_name):
     disabled in the config file, or ``None`` if the config file doesn't mention
     the given feature flag.
     """
-    setting_name = f"feature_flags.{feature_flag_name}"
-    return _bool_or_none_from_dict(request.registry.settings, setting_name)
+    return as_tristate(
+        request.registry.settings.get(f"feature_flags.{feature_flag_name}")
+    )
```

Comment:
This function is kind of doing what `as_tristate` is doing but also mixing in dict access.